parallel_cluster/conf/config

```diff
@@ -37,8 +65,9 @@ shared_dir       = efs
 encrypted        = false
 performance_mode = generalPurpose
 
-#[fsx lustrefs]
-#shared_dir                = /scratch
-#storage_capacity          = 3600
-#imported_file_chunk_size  = 1024
-#import_path               = s3://tothill-scratch
+[fsx lustrefs]
+shared_dir                = /fsx
+storage_capacity          = 3600
+imported_file_chunk_size  = 1024
+import_path               = s3://umccr-temp-dev/Alexis_parallel_cluster_test/input_folder
+export_path               = s3://umccr-temp-dev/Alexis_parallel_cluster_test/output_folder
```

Comment:
After our tests w/ Florian we decided to leave `export_path` out since it's just confusing (must use `hsm_upload` tool instead of just assuming that changes on the filesystem/bucket are transparent).

Reply:
What's the process then in order to map files back to the s3 buckets on closure of the cluster? Is there a way to place this under a separate mount point?

Reply:
@alexiswl On the AWS FSx web console, we found that the following checkbox combinations give the closest POSIX filesystem UX:

![Screen Shot 2020-08-11 at 2 18 15 pm](https://user-images.githubusercontent.com/175587/89856833-b8a07400-dbdd-11ea-95d6-2906ce62d1ef.png)

... that is, no weird export-specific commands, should just behave as a "close-to-normal" filesystem. There should not be separate mountpoints, just one.

Reply:
Related AWS (recent) announcement: https://aws.amazon.com/about-aws/whats-new/2020/07/amazon-fsx-lustre-supports-automatic-updates-amazon-s3/

Reply:
Reading the docs, does this mean we should be using the configuration that expects an existing filesystem instead?

See: https://docs.aws.amazon.com/parallelcluster/latest/ug/fsx-section.html#fsx-fs-id

Reply:
Yes, in my experience they should be created *a priori*, this is what I had to run on the instance to have them visible:

```
mount -t lustre -o noatime,flock fs-0a1c20d2dfb65e572.fsx.ap-southeast-2.amazonaws.com@tcp:/j6gqfbmv /scratch
mount -t lustre -o noatime,flock fs-0f042be802e974004.fsx.ap-southeast-2.amazonaws.com@tcp:/mwgafbmv /mnt/data
mount -t lustre -o noatime,flock fs-04acac0ac63c53a05.fsx.ap-southeast-2.amazonaws.com@tcp:/jlyafbmv /mnt/refdata
```

Both the part before the `@` as well as the ID coming after `:/` are generated when a FSx filesystem is created (via the webconsole or AWS CLI... in our case, preferably via CDK).

Reply:
I see, thanks! Yes, CDK might be the best bet for now - otherwise it's hard-coding them in the post_install args.
You didn't need to include these in the config at all?

Reply:
No, I didn't need them on the pcluster config on my tests: 

I defined the FSx filesystem via the web console, spawned the pcluster with `--norollback` and then ran the aforementioned mount commands (once the FSx filesystems were ready/created, they can take <5-6min to be ready).

Reply:
This should be fine, both the container and the /tmpdir for the cromwell job sit on the standard ebs volume. (Cromwell jobs generally put the tmp dir inside the folder for java based jobs but I put it a sed command in the sbatch script to modify the script to set to a root mount point).

Reply:
I think there would be a benefit to have the refdata entirely on SSD though, it's only a few GB so that shouldn't blow out costs right? Or is it a set cost to have the file system which is at minimum 1.2TBs?

Reply:
> I think there would be a benefit to have the refdata entirely on SSD though, it's only a few GB so that shouldn't blow out costs right? Or is it a set cost to have the file system which is at minimum 1.2TBs?

The latter, I had it running for just 2-3 days and it shoot up to >$50.

Reply:
Yep that looks okay then, from https://aws.amazon.com/fsx/lustre/pricing/ *For HDD-based file systems, the optional SSD cache improves performance by automatically placing your most frequently read data on SSD (the cache size is 20% of your file system size).* so all of the ref data would fit inside that 20%.*

However to get that 80% drop they've bragged about here, it does seem to be on the extreme lower end of the HDD use only.

The top HDD baseline (40MB/s/TiB) with 20% SSD cache compared to the lowest SSD-baseline (50MB/s) gives is only a 33% saving.
The lowest HDD baseline (12 MB/s/TiB) with 20% SSD cache compared to the lowest SSD-baseline (50MB/s) gives a 72 % saving.

Reading through this, it seems you do only pay for the filesystem when it's running, obvs storage in S3 that it's mounted to too. If the master instance is stopped, is the filesystem also terminated? I think that could be quite useful for someone that hasn't finished their work but would like to respawn. AFAIK the master instance isn't a spot instance and could be shutdown over night?

Reply:
> If the master instance is stopped, is the filesystem also terminated?

AFAIK, no, when you define the FSx via the web console, it goes `running` and that's totally independent from any EC2 instance, you can have 0 instances running and the FSx filesystem is "running", even if nothing uses it.

It is a pity that AWS does not charge for *USE* (as in I/O), like they do with S3. But it is understandable, the nature of FSx is essentially the same as EBS: a dedicated portion of a spinning drive (HDD) or memory ICs (SDD).

> AFAIK the master instance isn't a spot instance and could be shutdown over night?

No, [the pcluster `config` file is indeed SPOT](https://github.com/umccr/infrastructure/blob/master/parallel_cluster/conf/config#L16)... both master and compute are IIRC (when I looked at EC2 spot, there were 2 fulfilled spot requests every time I spawned the cluster).

Reply:
>  FSx filesystem is "running", even if nothing uses it.
Right, but it can be shut down and started upon request?

Reply:
> Right, but it can be shut down and started upon request?

Yes :)

Reply:
> No, the pcluster config file is indeed SPOT... both master and compute are IIRC

Looks like the master node can be shut down - suggests its not a spot instance.

![image](https://user-images.githubusercontent.com/8197659/90472649-5bb83700-e164-11ea-9bb7-68e659e813a9.png)

Reply:
Also lifecycle attribute is 'normal'